src/release/main.d

```diff
@@ -159,19 +159,58 @@ void main ( string[] params )
         if (readYesNoResponse("Would you like to close them now?"))
         {
             foreach (milestone; open_milestones)
-            {
-                writefln("Closing %s .. (%s open issues)",
-                         milestone.title, milestone.open_issues);
+                keepTrying(
+                {
+                    writef("Closing %s (%s open issues) ... ",
+                        milestone.title, milestone.open_issues);
 
-                updateMilestoneState(con, repo, milestone.number, State.closed);
-            }
+                    updateMilestoneState(con, repo, milestone.number, State.closed);
+                });
         }
     }
 
     writefln("All done.");
 }
 
 
+/*******************************************************************************
+
+    Repeatatly tries to call dg() if it threw an exception, while also keeping
+    the user informed about the attempts.
+
+    Works visually best if the dg() function prints it's actions according to
+    this pattern: "doing action abc ... " (no new line)
+
+    The function will then add either "success\n" or "failure: reason\n".
+
+    Params:
+        dg = delegate to call until it succeeds
+
+*******************************************************************************/
+
+void keepTrying ( void delegate ( ) dg )
+{
+    import core.thread;
+    import std.stdio;
+
+    while (true) try
```

Comment:
I would have a finite number of retries. Maybe ask the if you should keep trying after a few tries, but going infinite seems a bit extreme.